prov/efa/src/rxr/rxr_msg.c

```diff
@@ -74,15 +74,36 @@ ssize_t rxr_msg_post_cuda_rtm(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_ent
 	assert(tagged == 0 || tagged == 1);
 
 	delivery_complete_requested = tx_entry->fi_flags & FI_DELIVERY_COMPLETE;
-	if (tx_entry->total_len == 0) {
+	/*
+	 * Todo: use information in handshake packet to determine whether
+	 * the receiver supports gdrcopy.
+	 */
+	if (tx_entry->total_len == 0 || cuda_is_gdrcopy_enabled()) {
 		pkt_type = delivery_complete_requested ? RXR_DC_EAGER_MSGRTM_PKT : RXR_EAGER_MSGRTM_PKT;
-		return rxr_pkt_post_ctrl(rxr_ep, RXR_TX_ENTRY, tx_entry,
-					 pkt_type + tagged, 0, 0);
+		max_eager_data_size = rxr_pkt_req_max_data_size(rxr_ep,
+								tx_entry->addr,
+								pkt_type + tagged,
+								tx_entry->fi_flags, 0);
+
+		if (max_eager_data_size > rxr_env.efa_max_gdrcopy_msg_size)
```

Comment:
why not just do max_eager_data_size = MIN(max_eager_data_size, rxr_env.efa_max_gdrcopy_msg_size);

Reply:
used `MIN` in revision.